src/realm/error_codes.h

```diff
@@ -295,4 +295,16 @@ typedef enum realm_web_socket_errno {
     RLM_ERR_WEBSOCKET_FATAL_ERROR = 4405,
 } realm_web_socket_errno_e;
 
+typedef enum realm_sync_socket_callback_result {
+    // These error values are pulled directly from realm_errno_e
+    RLM_ERR_SYNC_SOCKET_SUCCESS = RLM_ERR_NONE,
+    RLM_ERR_SYNC_SOCKET_OPERATION_ABORTED = RLM_ERR_OPERATION_ABORTED,
+    RLM_ERR_SYNC_SOCKET_RUNTIME = RLM_ERR_RUNTIME,
+    RLM_ERR_SYNC_SOCKET_OUT_OF_MEMORY = RLM_ERR_OUT_OF_MEMORY,
```

Comment:
Are we sure we want/need users provide some of these errors? (i.e, RLM_ERR_SYNC_SOCKET_OUT_OF_MEMORY)

Reply:
No, but they could potentially be returned via the async write callback.